network/p2p/scoring/registry.go

```diff
@@ -403,17 +408,34 @@ func (r *GossipSubAppSpecificScoreRegistry) duplicateMessagesPenalty(pid peer.ID
 // OnInvalidControlMessageNotification is called when a new invalid control message notification is distributed.
 // Any error on consuming event must handle internally.
 // The implementation must be concurrency safe, but can be blocking.
+// Note: there is no real-time guarantee on processing the notification.
 func (r *GossipSubAppSpecificScoreRegistry) OnInvalidControlMessageNotification(notification *p2p.InvCtrlMsgNotif) {
+	lg := r.logger.With().Str("peer_id", p2plogging.PeerId(notification.PeerID)).Logger()
+	if ok := r.invCtrlMsgNotifWorkerPool.Submit(notification); !ok {
+		// we use a queue with a fixed size, so this can happen when queue is full or when the notification is duplicate.
+		// TODO: we have to add a metric for this case.
+		// TODO: we should not have deduplication for this case, as we need to penalize the peer for each misbehaviour, we need to add a nonce to the notification.
```

Comment:
are there issues for these?

Reply:
https://github.com/dapperlabs/flow-internal/issues/1910

Reply:
https://github.com/dapperlabs/flow-go/issues/6951
https://github.com/dapperlabs/flow-go/issues/6950